Frameworks/CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -1387,7 +1404,7 @@ CF_EXPORT CFStringRef CFURLCreateStringByAddingPercentEscapes(CFAllocatorRef all
             }
             else {
                 // FIXME: once CFString supports finding glyph boundaries walk by glyph boundaries instead of by unichars
-                if ( encoding == kCFStringEncodingUTF8 && CFCharacterSetIsSurrogateHighCharacter(ch) && idx + 1 < length && CFCharacterSetIsSurrogateLowCharacter(__CFStringGetCharacterFromInlineBufferQuick(&buf, idx+1)) ) {
```

Comment:
>kCFStringEncodingUTF8 [](start = 33, length = 21)

Please do not reformat CoreFoundation files; return line 1406 to its original state.